utils/loggers/wandb/wandb_utils.py

```diff
@@ -318,77 +399,123 @@ def log_model(self, path, opt, epoch, fitness_score, best_model=False):
-        if data.get('val'):
-            data['val'] = WANDB_ARTIFACT_PREFIX + str(Path(project) / 'val')
+            self.train_artifact = (
+                self.create_dataset_table(
+                    LoadImagesAndLabels(data["train"], rect=True, batch_size=1),
+                    names,
+                    name="train",
+                )
+                if data.get("train")
+                else None
+            )
+            if data.get("train"):
+                data["train"] = (
+                    WANDB_ARTIFACT_PREFIX
+                    + str(Path(entity) / Path(project) / "train")
+                    + ":"
+                    + alias
+                )
+
+        self.val_artifact = (
+            self.create_dataset_table(
+                LoadImagesAndLabels(data["val"], rect=True, batch_size=1),
+                names,
+                name="val",
+            )
+            if data.get("val")
+            else None
+        )
+        if data.get("val"):
+            data["val"] = (
+                WANDB_ARTIFACT_PREFIX
+                + str(Path(entity) / Path(project) / "val")
+                + ":"
+                + alias
+            )
 
         path = Path(data_file)
         # create a _wandb.yaml file with artifacts links if both train and test set are logged
         if not log_val_only:
-            path = (path.stem if overwrite_config else path.stem + '_wandb') + '.yaml'  # updated data.yaml path
-            path = ROOT / 'data' / path
-            data.pop('download', None)
-            data.pop('path', None)
-            with open(path, 'w') as f:
+            path = (
+                path.stem if overwrite_config else path.stem + "_wandb"
+            ) + ".yaml"  # updated data.yaml path
+            path = ROOT / "data" / path
+            data.pop("download", None)
+            data.pop("path", None)
+            with open(path, "w") as f:
                 yaml.safe_dump(data, f)
                 LOGGER.info(f"Created dataset config file {path}")
 
-        if self.job_type == 'Training':  # builds correct artifact pipeline graph
+        if self.job_type == "Training":  # builds correct artifact pipeline graph
             if not log_val_only:
                 self.wandb_run.log_artifact(
-                    self.train_artifact)  # calling use_artifact downloads the dataset. NOT NEEDED!
+                    self.train_artifact
+                )  # calling use_artifact downloads the dataset. NOT NEEDED!
             self.wandb_run.use_artifact(self.val_artifact)
             self.val_artifact.wait()
-            self.val_table = self.val_artifact.get('val')
+            self.val_table = self.val_artifact.get("val")
             self.map_val_table_path()
         else:
             self.wandb_run.log_artifact(self.train_artifact)
             self.wandb_run.log_artifact(self.val_artifact)
-        return path
+        return str(path)
```

Comment:
This `Path` object needs to be converted to `str` since line 56 checks `data_files.endswith('yaml')`